src/lib/model/InboundTransfersModel.js

```diff
@@ -370,6 +315,356 @@ class InboundTransfersModel {
+        try {
+            // retrieve bulk quote data
+            const bulkQuote = await this._cache.get(`bulkQuotes_${bulkPrepareRequest.bulkTransferId}`);
+
+            if (!bulkQuote) {
+                // Check whether to allow transfers without a previous quote.
+                if (!this._allowTransferWithoutQuote) {
+                    throw new Error(`Corresponding bulk quotes not found for bulk transfers ${bulkPrepareRequest.bulkTransferId}`);
+                }
+            }
+
+            // create an index of individual quote results indexed by transactionId for faster lookups
+            const quoteResultsByTrxId = {};
+
+            if (bulkQuote && bulkQuote.mojaloopResponse && bulkQuote.mojaloopResponse.individualQuoteResults) {
+                for (const quoteResult of bulkQuote.mojaloopResponse.individualQuoteResults) {
+                    quoteResultsByTrxId[quoteResult.transactionId] = quoteResult;
+                }
+            }
+
+            // transfer fulfilments
+            const fulfilments = {};
+
+            // collect errors for each transfer
+            let individualTransferErrors = [];
+
+            // validate individual transfer
+            for (const transfer of bulkPrepareRequest.individualTransfers) {
+                // decode ilpPacked for this transfer to get transaction object
+                const transactionObject = this._ilp.getTransactionObject(transfer.ilpPacket);
+
+                // we use the transactionId from the decoded ilpPacked in the transfer to match a corresponding quote
+                const quote = quoteResultsByTrxId[transactionObject.transactionId] || null;
+
+                // calculate or retrieve fulfilments and conditions
+                let fulfilment = null;
+                let condition = null;
+
+                if (quote) {
+                    fulfilment = bulkQuote.fulfilments[quote.quoteId];
+                    condition = quote.condition;
+                }
+                else {
+                    fulfilment = this._ilp.calculateFulfil(transfer.ilpPacket);
+                    condition = this._ilp.calculateConditionFromFulfil(fulfilment);
+                }
+
+                fulfilments[transfer.transferId] = fulfilment;
+
+                // check incoming ILP matches our persisted values
+                if (this._checkIlp && (transfer.condition !== condition)) {
+                    const transferError = this._handleError(new Error(`ILP condition in bulk transfers prepare for ${transfer.transferId} does not match quote`));
+                    individualTransferErrors.push({ transferId: transfer.transferId, transferError });
+                }
+            }
+
+            if (bulkQuote && this._rejectTransfersOnExpiredQuotes) {
+                const now = new Date().toISOString();
+                const expiration = bulkQuote.mojaloopResponse.expiration;
+                if (now > expiration) {
```

Comment:
@oderayi are you sure this is a valid comparison? are both these vars Date objects?

Reply:
@bushjames The two variables were in the simplified ISO 8601 format and so were comparable. I have however converted both to Date objects before comparing being the recommended JavaScript style.